kaspresso/src/main/kotlin/com/kaspersky/kaspresso/systemsafety/SystemDialogSafetyProviderImpl.kt

```diff
@@ -111,7 +111,11 @@ class SystemDialogSafetyProviderImpl(
      */
     private fun isAndroidSystemDetected(): Boolean {
         with(uiDevice) {
-            var isSystemDialogVisible = SystemDialogSafetyPattern.values().any { isVisible(By.pkg(it.pattern).clazz(FrameLayout::class.java)) }
+            var isSystemDialogVisible = if (systemDialogsSafetyParams.shouldIgnorePermissionController) {
+                SystemDialogSafetyPattern.values().filter { it != SystemDialogSafetyPattern.PERMISSION_API30 }.any { isVisible(By.pkg(it.pattern).clazz(FrameLayout::class.java)) }
```

Comment:
I think it would be better to generalize this solution to all permissions dialogs. Blocking only permission_controller package is not very intuitive. Please, rename shouldIgnorePermissionController to shouldIgnorePermissionDialogs or similar and compare with both SystemDialogSafetyPattern.PERMISSION_API30 and SystemDialogSafetyPattern.PERMISSION_API23 on this line

Reply:
done